assets/src/dashboard/components/typeahead/useTypeahead.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useEffect, useMemo, useState, useRef } from 'react';
+
+export default function useTypeahead({ items, value }) {
+  const [showMenu, setShowMenu] = useState(false);
+  const [menuFocused, setMenuFocused] = useState(false);
+  const [inputValue, setInputValue] = useState(value);
+  const [selectedValueIndex, setSelectedValueIndex] = useState(-1);
+
+  const menuOpened = useRef(false);
+
+  const isMenuOpen = useMemo(() => {
+    return showMenu && items.length > 0 && inputValue.length > 0;
+  }, [items, showMenu, inputValue]);
+
+  const isInputExpanded = useMemo(() => {
+    return menuFocused || inputValue.length > 0;
+  }, [menuFocused, inputValue]);
```

Comment:
Same here

Reply:
thanks for catching these - i totally glossed over this and just moved it without noticing it's unnecessary.